pkg/elle/core/graph.go

```diff
@@ -19,75 +19,263 @@ type DirectedGraph struct {
 
 // Vertices returns the set of all vertices in graph
 func (g *DirectedGraph) Vertices() []Vertex {
-	panic("impl me")
+	vertices := []Vertex{}
+
+	for key := range g.Outs {
+		vertices = append(vertices, key)
+	}
+	return vertices
 }
 
-// In returns inbound edges to v in graph g
+// In returns inbound vertices to v in graph g
 func (g *DirectedGraph) In(v Vertex) []Vertex {
-	panic("impl me")
+	_, ok := g.Ins[v]
+	if !ok {
+		return []Vertex{}
+	} else {
+		return g.Ins[v]
+	}
 }
 
-// Out returns outbound edges from v in graph g
+// Out returns outbound vertices from v in graph g
 func (g *DirectedGraph) Out(v Vertex) []Vertex {
-	panic("impl me")
+	_, ok := g.Outs[v]
+	if !ok {
+		return []Vertex{}
+	}
+
+	vertices := []Vertex{}
+
+	for key := range g.Outs[v] {
+		vertices = append(vertices, key)
+	}
+	return vertices
 }
 
 // Edges returns the edge between two vertices
 func (g *DirectedGraph) Edges(a, b Vertex) []Edge {
-	panic("impl me")
+	_, oka := g.Outs[a]
+	_, okb := g.Ins[b]
```

Comment:
I think you only need to check `g.Outs[a]` and `g.Outs[a][b]` exist, `g.Ins` is used for look for reverse paths.